src/pages/home/components/Sidebar/Sidebar.tsx

```diff
@@ -9,37 +9,35 @@ import SearchClue from '~components/SearchClue/SearchClue';
 
 import scss from './Sidebar.module.scss';
 
-function Sidebar() {
+function Sidebar(): ReactNode {
   const [value, setValue] = useState('');
-  const [isOpen, setOpen] = useState(false);
+  const [isOpen, setIsOpen] = useState(false);
   const inputSearchRef = useRef<TInputComponent | null>(null);
 
-  const handleToggleModal = () => {
-    setOpen((isOpen) => !isOpen);
+  const handleToggleModal = (): void => {
+    setIsOpen((currentIsOpen) => !currentIsOpen);
   };
 
-  const renderSearchClue = () => {
-    return (
-      <SearchClue
-        clearInput={inputSearchRef.current?.clearInput}
-        value={value}
-        label="Searching for"
-      />
-    );
-  };
+  const renderSearchClue = (): ReactNode => (
+    <SearchClue
+      clearInput={inputSearchRef.current?.clearInput}
+      label="Searching for"
+      value={value}
+    />
+  );
 
   return (
-    <AccordionTab title="Select Social Media" hideCloseButton>
+    <AccordionTab hideCloseButton title="Select Social Media">
       <div className={scss.content}>
         <InputSearch
           className={scss.searchBar}
-          onChange={(value) => setValue(value as string)}
+          error={false}
+          onChange={(newValue) => setValue(newValue as string)}
           placeholder="Search for social media"
           ref={inputSearchRef}
-          error={false}
         />
 
-        {!value && renderSearchClue()}
+        {value ? null : renderSearchClue()}
```

Comment:
Pqqqqqq???? 😭😭

Reply:
quando eu deixo da antiga forma aponta esse erro Potential leaked value that might cause unintentionally rendered values or rendering crashes

Reply:
As formas que pensei foi 
const renderSearchClue = (): ReactElement | null =>

{renderSearchClue()}


const renderSearchClue = (): ReactElement => {
  if (!value) {
    return (
      <SearchClue
        clearInput={inputSearchRef.current?.clearInput}
        label="Searching for"
        value={value}
      />
    );
  }
  return <></>; // Retorna um fragmento React vazio
};

se nao quiser vazio pode ser div mesmo ou span

Reply:
> quando eu deixo da antiga forma aponta esse erro Potential leaked value that might cause unintentionally rendered values or rendering crashes

Esse problema de lint é pq ele acha que value é number, mas é string, literalmente nunca vai dar pau sendo string.

No caso ficaria `value && renderSearchClue()`

Se a regra de lint ta apontando problema nesse caso (considerando o `value` string) a regra precisa ser ajustada pra fazer esse tipo de obrigatoriedade apenas em number